Sources/Purchasing/Offering.swift

```diff
@@ -165,6 +173,21 @@ import Foundation
 
 }
 
+extension Offering {
+
+    /**
+     - Returns: the `metadata` value associated to `key` for the expected type,
+     or `defaultValue` if not found, or it's not the expected type.
+     */
+    public func getMetadataValue<T>(for key: String, default: T) -> T {
```

Comment:
Would this also support null default values? Like `let metadataInt: Int? = offering.getMetadataValue(for: "key", default: nil)`. I think that could be useful.

Reply:
It should be let me write another test for that! 😊

Reply:
Yup, this works!

```swift
let optionalInt: Int? = offeringA.getMetadataValue(for: "optionalInt", default: nil)
expect(optionalInt).to(beNil())
```